Automattic/wp-calypso
# A4A: Implement the 'Add site via Jetpack connection' modal.

This pull request implements the modal for adding a site via Jetpack connection, which will be used in the central hub site selector.

<img width="817" alt="Screenshot 2024-07-02 at 1 57 29 AM" src="https://github.com/Automattic/wp-calypso/assets/56598660/8824d286-bdf2-44ed-9226-2412c747f2e8">

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/679

## Proposed Changes

* Implement the `A4AThemedModal` component as a reusable skeleton for a themed-based modal.
* Implement the `AddSiteViaJetpackConnectionModal` component based on the new reusable themed-based modal.


## Testing Instructions

* Use the A4A live link and go to `/overview` page.
* Click the **Add site** dropdown menu at the top right.
* Click the **Via Jetpack** option to show the modal.
* Test that the modal functions as expected.

## Pre-merge Checklist
